### ISSUE TITLE
[Bug]: Signature validation is failing during tag releases

### ISSUE THREAD

**Issue by martincostello** [App-vNext/Polly#1760]
### Describe the bug

Trying to release 8.1.0, the validate-signed-packages job is failing as the Authenticode signatures appear to be invalid: [workflow logs](https://github.com/App-vNext/Polly/actions/runs/6705480558).

I've unpublished the release and deleted the tag while I investigate what's going wrong.

### Expected behavior

The binaries are signed with a valid Authenticode signature.

### Actual behavior

Multiple binaries are not correctly Authenticode signed.

### Steps to reproduce

Create a new tag.

### Exception(s) (if any)

See [workflow logs](https://github.com/App-vNext/Polly/actions/runs/6705480558/job/18220705231#step:6:76).

**Comment by martincostello** [comment]
Looks like there was a new release of the [sign tool](https://www.nuget.org/packages/sign/0.9.1-beta.23530.1) yesterday - maybe updating that will help.

**Comment by martincostello** [comment]
Investigation so far:

1. Trying to view the Authenticode signatures of the DLLs inside the NuGet packages in NuGet Package Explorer fails with this exception: 
![image](https://github.com/App-vNext/Polly/assets/1439341/2c91e1e0-5407-4206-9e2f-787484ce779e)
1. Running [vcsjones/AuthenticodeLint](https://github.com/vcsjones/AuthenticodeLint) locally fails with a similar exception [here](https://github.com/vcsjones/AuthenticodeLint/blob/ccfaec53ee5c1b14f029cb8156e0653c530f8b65/AuthenticodeLint/SignatureExtensions.cs#L10): 
![image](https://github.com/App-vNext/Polly/assets/1439341/0bb510c3-75d1-4bb1-ae5b-a5441b2c04d2)
```
System.ArgumentNullException
  HResult=0x80004003
  Message=Value cannot be null. Arg_ParamName_Name
  Source=System.Private.CoreLib
  StackTrace:
   at System.Runtime.InteropServices.Marshal.CopyToManaged[T](IntPtr source, T[] destination, Int32 startIndex, Int32 length)
   at AuthenticodeExaminer.CmsSignatureBase.ReadAttributes(CRYPT_ATTRIBUTES attributes)
   at AuthenticodeExaminer.CmsSignature.InitFromHandles(CryptMsgSafeHandle messageHandle, LocalBufferSafeHandle signerHandle)
   at AuthenticodeExaminer.CmsSignature..ctor(AsnEncodedData data, SignatureKind kind)
   at AuthenticodeExaminer.CmsSignature.GetNestedSignatures()
   at AuthenticodeLint.SignatureExtensions.<VisitAll>d__0.MoveNext() in C:\Coding\vcsjones\AuthenticodeLint\AuthenticodeLint\SignatureExtensions.cs:line 10
   at AuthenticodeLint.SignatureExtensions.<VisitAll>d__1.MoveNext() in C:\Coding\vcsjones\AuthenticodeLint\AuthenticodeLint\SignatureExtensions.cs:line 38
   at AuthenticodeLint.Rules.NoWeakFileDigestAlgorithmsRule.Validate(IReadOnlyList`1 graph, SignatureLogger verboseWriter, CheckConfiguration configuration) in C:\Coding\vcsjones\AuthenticodeLint\AuthenticodeLint\Rules\10002-NoWeakFileDigestAlgorithmsRule.cs:line 20
   at AuthenticodeLint.CheckEngine.RunAllRules(String file, IReadOnlyList`1 signatures, List`1 collectors, CheckConfiguration configuration) in C:\Coding\vcsjones\AuthenticodeLint\AuthenticodeLint\CheckEngine.cs:line 59
   at AuthenticodeLint.Program.Main(String[] args) in C:\Coding\vcsjones\AuthenticodeLint\AuthenticodeLint\Program.cs:line 175
```

Maybe something in the last two .NET SDK updates broke something in NuGet when packages are signed or something in Azure Sign Service broken something...

**Comment by martincostello** [comment]
Rebuilding the linter from source and building its dependency from source works as expected and validates the signatures.

Short-term fix is to commit that into a fork of the repository and to build and run that version in our release process. Once that's set up and working, I can dig further post-release to find the true root cause of the issue and sort things out in a less string-and-duct-tape way.

**Comment by martincostello** [comment]
Possibly somehow related to https://github.com/dotnet/sdk/issues/33928.

**Comment by martincostello** [comment]
Resolved by #1761 - looks like the code issue is https://github.com/vcsjones/AuthenticodeExaminer/issues/19, which also affects NuGet Package Explorer, but _something else_ changed to trigger that underlying issue.
